src/vm/comutilnative.cpp

```diff
@@ -2624,14 +2626,107 @@ FCIMPL6(INT32, ManagedLoggingHelper::GetRegistryLoggingValues, CLR_BOOL* bLoggin
 }
 FCIMPLEND
 
+static BOOL HasOverriddenEqualsWorker(MethodTable* mt)
+{
+    MethodTable::IntroducedMethodIterator it(mt);
+    for (; it.IsValid(); it.Next())
+    {
+        if (strcmp(it.GetMethodDesc()->GetName(), "Equals") == 0)
```

Comment:
@jkotas that code sounds like not comparing the signature of the method. I'll take a look to see if it can be applied to `Equals` and `GetHashCode`